ingenialink/ethercat/servo.py

```diff
@@ -320,6 +324,75 @@ def set_pdo_watchdog_time(self, timeout: float) -> None:
             self.ETHERCAT_PDO_WATCHDOG, self.SECONDS_TO_MS_CONVERSION_FACTOR * timeout
         )
 
+    def _read_esc_eeprom(
+        self,
+        address: int,
+        length: int = DEFAULT_EEPROM_READ_BYTES_LENGTH,
+        timeout: int = DEFAULT_EEPROM_OPERATION_TIMEOUT_uS,
+    ) -> bytes:
+        """Read from the ESC EEPROM.
+
+        Args:
+            address: EEPROM address to be read.
+            length: Length of data to be read. By default, 2 bytes are read.
+            timeout: Operation timeout (microseconds). By default, 200.000 us.
+
+        Returns:
+            EEPROM data. The read data.
+
+        Raises:
+            ValueError: If the length to be read has an invalid value.
+
+        """
+        if length < 1:
+            raise ValueError("The minimum length is 1 byte.")
+        data = bytes()
+        start_address = address
+        while len(data) < length:
+            data += self.slave.eeprom_read(start_address, timeout)
+            start_address += 2
+        if len(data) > length:
+            data = data[:length]
+        return data
+
+    def _write_esc_eeprom(
+        self, address: int, data: bytes, timeout: int = DEFAULT_EEPROM_OPERATION_TIMEOUT_uS
+    ) -> None:
+        """Write to the ESC EEPROM.
+
+        Args:
+            address: EEPROM address to be written.
+            data: Data to be written. The data length must be a multiple of 2 bytes.
+            timeout: Operation timeout (microseconds). By default, 200.000 us.
+
+        Raises:
+            ValueError: If the data has the wrong size.
+
+        """
+        if len(data) % 2 != 0:
+            raise ValueError("The data length must be a multiple of 2 bytes.")
+        start_address = address
+        while data:
+            self.slave.eeprom_write(start_address, data[:2], timeout)
+            data = data[2:]
```

Comment:
There are ways to iterate over array 2 by 2 that are o(1) instead of this that is o(2)

Reply:
Essentially the same.
What do you suggest?